packages/test-runner/src/fixtures.ts

```diff
@@ -185,21 +184,28 @@ export class FixturePool {
   }
 }
 
-export function fixturesForCallback(callback: Function): string[] {
-  const names = new Set<string>();
-  const visit  = (callback: Function) => {
-    for (const name of fixtureParameterNames(callback)) {
-      if (name in names)
-        continue;
-      names.add(name);
+function getRegistration(name: string) {
+  const registration = registrations.get(name);
```

Comment:
```suggestion
  // It might seem that the below code is not correct, but we believe it is.
  // For example, consider the case where we run testA.spec.ts and testB.spec.ts in the same worker.
  // testA uses fixture "page" and testB overrides and uses fixture "page". Since we `rerun` test fixtures
  // in the order of the require traversal, correct fixtures will be "on top" of the `registrations` map and will
  // be looked up.

  // And it does not matter for the worker level fixtures because complete worker fixture registration tree is
  // captured in the worker hash, hence we know that all the tests in the same worker will share that fingerprint
  // and that the latest fixture will be "on top" of the map.
  const registration = registrations.get(name);
```